ql/src/java/org/apache/hadoop/hive/ql/exec/tez/TezProcessor.java

```diff
@@ -293,6 +293,13 @@ protected void initializeAndRunProcessor(Map<String, LogicalInput> inputs,
       rproc.run();
 
       perfLogger.perfLogEnd(CLASS_NAME, PerfLogger.TEZ_RUN_PROCESSOR);
+
+      // Try to call canCommit to AM. If there is no other speculative attempt execute canCommit, then continue.
+      // If there are other speculative attempt execute canCommit first, then wait until the attempt is killed
+      // or the committed task fails.
+      while (!this.processorContext.canCommit()) {
```

Comment:
can we drop this `processorContext` global var as it seems unnecessary and confusing?

Reply:
It seems that it doesn't matter. If this is the case, I also removed the processorContext parameter passing to avoid unnecessary use of the stack.
However, the large amount of code modified in this PR has nothing to do with this PR. I don’t know if this is a good idea?